x/twap/api_test.go

```diff
@@ -11,26 +11,43 @@ import (
 )
 
 var (
+	// when creating twap inputs, we use a function that, when set to true, returns the
+	// base asset as the lexicographically smaller denom and the quote as the larger. When
+	// set to false, this order is switched. These constants are provided to understand the
+	// base/quote asset for every test at a glance rather than a raw boolean value.
+	baseQuoteAB, baseQuoteCA, baseQuoteBC = true, true, true
+	baseQuoteBA, baseQuoteAC, baseQuoteCB = false, false, false
+
 	ThreePlusOneThird sdk.Dec = sdk.MustNewDecFromStr("3.333333333333333333")
 
 	// base record is a record with t=baseTime, sp0=10(sp1=0.1) accumulators set to 0
 	baseRecord types.TwapRecord = newTwapRecordWithDefaults(baseTime, sdk.NewDec(10), sdk.ZeroDec(), sdk.ZeroDec())
 
-	// accum0 = 10 seconds * (spot price = 10) = OneSec * 10 * 10
-	// accum1 = 10 seconds * (spot price = 0.1) = OneSec
-	accum0, accum1 sdk.Dec = OneSec.MulInt64(10 * 10), OneSec
+	tapRecordAB, tapRecordAC, tapRecordBC = newThreeAssetPoolTwapRecordWithDefaults(
+		baseTime, sdk.NewDec(10), sdk.ZeroDec(), sdk.ZeroDec(), sdk.ZeroDec())
+
+	// accumA = 10 seconds * (spot price = 10) = OneSec * 10 * 10
+	// accumB = 10 seconds * (spot price = 0.1) = OneSec
+	// accumC = 10 seconds * (spot price = 20) = OneSec * 10 * 20
+	accumA, accumB, accumC sdk.Dec = OneSec.MulInt64(10 * 10), OneSec, OneSec.MulInt64(10 * 20)
 
 	// accumulators updated from baseRecord with
 	// t = baseTime + 10
-	// sp0 = 5, sp1 = 0.2
+	// spA = 5, spB = 0.2, spC = 10
 	tPlus10sp5Record = newTwapRecordWithDefaults(
-		baseTime.Add(10*time.Second), sdk.NewDec(5), accum0, accum1)
+		baseTime.Add(10*time.Second), sdk.NewDec(5), accumA, accumB)
+
+	tPlus10sp5TapRecordAB, tPlus10sp5TapRecordAC, tPlus10sp5TapRecordBC = newThreeAssetPoolTwapRecordWithDefaults(
+		baseTime.Add(10*time.Second), sdk.NewDec(5), accumA, accumB, accumC)
 
 	// accumulators updated from tPlus10sp5Record with
 	// t = baseTime + 20
-	// sp0 = 2, sp1 = 0.5
+	// spA = 2, spB = 0.5, spC = 4
 	tPlus20sp2Record = newTwapRecordWithDefaults(
 		baseTime.Add(20*time.Second), sdk.NewDec(2), OneSec.MulInt64(10*10+5*10), OneSec.MulInt64(3))
```

Comment:
nit: to reduce cognitive overhead
```suggestion
	tPlus20sp2Record = newTwapRecordWithDefaults(
		baseTime.Add(20*time.Second),
		sdk.NewDec(2),               // spot price 0
		OneSec.MulInt64(10*10+5*10), // accum A
		OneSec.MulInt64(3))          // accum B
```